SEBrowser/Controllers/OpenXDAController.cs

```diff
@@ -219,32 +225,47 @@ private string getEventTypeFilter(EventSearchPostData postData)
 
         private string getPhaseFilter(EventSearchPostData postData)
         {
-            List<string> phase = new List<string>();
-
-            if (postData.PhaseA && postData.PhaseB && postData.PhaseC)
-                return "";
-            if (!postData.PhaseA && !postData.PhaseB && !postData.PhaseC)
+            Dictionary<string, bool> phases = new Dictionary<string, bool> 
+            {
+                ["AN"] = postData.PhaseAN,
+                ["BN"] = postData.PhaseBN,
+                ["CN"] = postData.PhaseCN,
+                ["AB"] = postData.PhaseAB,
+                ["BC"] = postData.PhaseBC,
+                ["ABG"] = postData.PhaseABG,
+                ["BCG"] = postData.PhaseBCG,
+                ["ABC"] = postData.PhaseABC,
+                ["ABCG"] = postData.PhaseABCG
+            };
+
+            bool allTrue = true;
+            bool allFalse = true;
+            foreach (var value in phases.Values) 
+            {
+                if (value)
+                {
+                    allFalse = false;
+                }
+                else
+                {
+                    allTrue = false;
+                }
+            }
+            if (allTrue || allFalse)
+            {
                 return "";
-
-            if (postData.PhaseA)
-            {
-                phase.Add("(SELECT Phase.Name FROM Disturbance LEFT JOIN Phase ON Disturbance.PhaseID = Phase.ID WHERE Disturbance.ID = EventWorstDisturbance.WorstDisturbanceID) IN ('AN','AB','CA')");
-                phase.Add("FaultSummary.FaultType IN ('AN','AB','CA', 'ABG', 'CAG', 'ABC', 'ABCG')");
-            }
-            if (postData.PhaseB)
-            {
-                phase.Add("(SELECT Phase.Name FROM Disturbance LEFT JOIN Phase ON Disturbance.PhaseID = Phase.ID WHERE Disturbance.ID = EventWorstDisturbance.WorstDisturbanceID) IN ('BN','AB','BC')");
-                phase.Add("FaultSummary.FaultType IN ('BN','AB','BC', 'ABG', 'BCG', 'ABC', 'ABCG')");
-            }
-            if (postData.PhaseC)
-            {
-                phase.Add("(SELECT Phase.Name FROM Disturbance LEFT JOIN Phase ON Disturbance.PhaseID = Phase.ID WHERE Disturbance.ID = EventWorstDisturbance.WorstDisturbanceID) IN ('CN','BC','CA')");
-                phase.Add("FaultSummary.FaultType IN ('CN','BC','CA', 'BCG', 'CAG', 'ABC', 'ABCG')");
             }
 
-
-
-            return string.Join(" OR ", phase);
+            string phaseCombined = "";
+            foreach (var item in phases) 
+            {
+                if (item.Value) 
+                {
+                    phaseCombined += ("\'" + item.Key + "\', ");
+                }
+            }
```

Comment:
use LINQ to simplify something like `string.Join(phases.Where(item => item.Value).Select(item => "\'" + item.Key + "\'"),",")`